target/linux/pistachio/base-files/etc/diag.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+#
+# Copyright (C) 2017 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+. /lib/functions/leds.sh
+. /lib/pistachio.sh
+
+status_led="marduk:red:heartbeat"
+
+set_state() {
+
+	case "$1" in
+	preinit)
+		status_led_blink_preinit
+		;;
+	failsafe)
+		status_led_blink_failsafe
+		;;
+	preinit_regular)
+		status_led_blink_preinit_regular
+		;;
+	upgrade)
+		status_led_blink_slow
+		;;
+	done)
+		status_led_set_heartbeat
```

Comment:
My idea (more like a long-term plan) was (still is, probably) to move out state-related status LED setup code from all target specific "diag.sh" files to a common place/function and then eventually allow override/extend defaults in target (we do other things per state, [like in ar71xx](https://github.com/lede-project/source/blob/master/target/linux/ar71xx/base-files/etc/diag.sh#L492)).

At now we already fill "diag.sh" file per (new) target by copy&paste same file from other place/s and in almost all (alive/maintained) targets "set_state()" function has the exact same or very similar content (`grep -rn "set_state()" target/linux/*/base-files/ -A999`). This approach doesn't seem correct for me. If we include some new, project-wide state (in fact, I have here something in mind: firstboot triggered inside failsafe), we would need to include same change in all target specific "diag.sh" files.

Anyway, I really don't know the reason/s, why by default status LED is set to "on" (and not to a "heartbeat" for example) in "done" state (this seems to be a general rule, across all targets, excluding old and almost forgotten ar7 which I want to "fix"), but I'm pretty sure it's done this way not without a reason. Maybe because of some historical reasons or just because we use "blinking" patterns during all "preceding" states (failsafe, preinit, etc.) and "heartbeat" might be confusing (I wasn't able to find any better argument here :confused:).

IMHO, this topic should be discussed in larger group and lede-dev mailing list looks like a correct place for it.

> I am happy to drop the use of status_led_blink_slow if it helps.

At least in "ramips" and "ipq806x" targets we use for "upgrade" state the same pattern as for "preinit_regular" ([ramips/base-files/etc/diag.sh#L271](https://github.com/lede-project/source/blob/master/target/linux/ramips/base-files/etc/diag.sh#L271)).

> If there are major concerns I can change this but believe there are valid reasons for keeping it. Also just because LEDE isn't using that function doesn't mean there aren't any forks that do make full use of it.

Personally, it doesn't make sense for me to keep functions which are not used at all, across whole code. If some fork/s make use of it, they can add them in fork as well.